scripts/testing/create_override_csv.py

```
@@ -0,0 +1,132 @@
+
+For example, if we have keys one, two, three, we'd have files overload_all.csv,
+overload_one.csv, overload_two.csv, overload_three.csv, overload_one_two.csv,
+overload_one_three.csv, overload_two_three.csv and overload_one_two_three.csv.
+
+The final overload file will be a concatanation of any files found with those
+combinations, prioritising the least number of keys first. It will also search
+all directories passed in for each key combination.
+"""
+
+import argparse
+import os
+import sys
+
+
+def get_args():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        "-k",
+        "--key", nargs="+", default=[],
+        help="list of keys")
+    parser.add_argument(
+        "-d",
+        "--dir", nargs="+", default=['.'],
+        help="list of dirs")
+    parser.add_argument(
+        "-o",
+        "--output", default='override_combined.csv',
+        type=str,
+        help="output file (default override_combined.csv)")
+
+    parser.add_argument(
+        "-v",
+        "--verbose",
+        action="count",
+        default=0,
+        help="Make the output more verbose, -v shows csv files, -vv also cats the final output")
+    args = parser.parse_args()
+    return args
+
+
+def make_combos(keys, base, file_list, level):
+    """ Create combinations of keys for a particular level upwards.
+
+    Keyword arguments:
+        keys      -- List of keys to combine, suitable for the base
+        base      -- base filename to extend
+        file_list -- List of lists of filenames, one list for each level
+        level     -- Current level
+
+    We want to create combinations of keys based on maintaining the order passed
+    in and combining with any keys which are beyond them in the list. Also we
+    want to make combinations of different numbers of keys each at a distinct
+    level. e.g. if we have keys one,two,three, we'd have keys L 1: one, two,
+    three, L 2: one_two, one_three, two_three L 3: one_two_three. make_combos
```

I think this means we can, depending on how we call this, have either e.g. `override_opencl_x86_64.csv` or `override_x86_64_opencl.csv`, but not both, right? That's probably fine but given that it silently ignores files that have the wrong name, I think we should be very clear about what we expect.

---

yes the order is important - what are you suggesting? I probably could make it so it considers all orders or perhaps you just mean make it clearer in the documentation?